### PR TITLE
Implemented a method to change the data of packet.

### PR DESCRIPTION
**Pull request by renatocarusos**
This method work with char[], string, int or float variables. In float, there have an option to determine how many decimal will be, instead as three, as default.
There is an example sketch too.

### CODE REVIEW

File: src/Packet.cpp [hunk 1]
```diff
@@ -151,6 +151,44 @@ void Packet::copyInto(Packet& pack) {
...
 }
 
+void Packet::setPacketData(char charArray[240]){
+  strcpy(data, charArray);
+}
+
+void Packet::setPacketData(String wordy){
+  char charArray[240];
+  wordy.toCharArray(charArray, sizeof(wordy));
+  this->setPacketData(charArray);
```
Comment (jgromes) [inline src/Packet.cpp]: Unneccessary `this` pointer (it is implied when calling member method `setPacketData` from inside its own class).
The same goes for all `this` pointers in this file.

File: src/Packet.cpp [hunk 2]
```diff
@@ -151,6 +151,44 @@ void Packet::copyInto(Packet& pack) {
...
+    char charArray[240];
+    itoa(i, charArray, 10);
+    this->setPacketData(charArray);
+  } else {
+    String floatString = String(f, _decimals);
+    this->setPacketData(floatString);
+  }
+}
+
+void Packet::setPacketData(float f, uint8_t decimals) {
```
Comment (jgromes) [inline src/Packet.cpp]: This is not a good way to overload functions - you have basically the same code twice here in both implementations of `setPacketData`.

Delete the previous `setPacketData` (line 164) and in this one, add default value to the `decimals` argument in Packet.h

File: src/Packet.h [hunk 3]
```diff
@@ -35,6 +35,11 @@ class Packet {
     
     void copyInto(Packet& pack);
     
+    void setPacketData(char charArray[240]);
+    void setPacketData(String wordy);
+    void setPacketData(float f);
+    void setPacketData(float f, uint8_t decimals);
```
Comment (jgromes) [inline src/Packet.h]: Change to `void setPacketData(float f, uint8_t decimals = 3);`

File: src/Packet.cpp [hunk 4]
```diff
@@ -151,6 +151,44 @@ void Packet::copyInto(Packet& pack) {
   strcpy(pack.data, data);
 }
 
+void Packet::setPacketData(char charArray[240]){
+  strcpy(data, charArray);
+}
+
+void Packet::setPacketData(String wordy){
```
Comment (jgromes) [inline src/Packet.cpp]: Change the argument to `String str`.

**Review by jgromes — CHANGES_REQUESTED** [review]
Nice, I was thinking of implementing something similar myself, just never got around to actually doing it. Before I merge this, there's a few minor issues, I would be glad if you could address them.

I would like to also rewrite the example code, but I will do that myself after the merge (so that the writing style is consistent across all the examples).

Thanks!

**Comment by renatocarusos** [comment]
I was really worried about the style of the sample code. But I figured you'd stylize it according to your sketch.

**Review by jgromes — APPROVED** [review]

**Pull request merged**